src/native/libs/configure.cmake

```diff
@@ -1101,6 +1101,19 @@ check_symbol_exists(
     grp.h
     HAVE_GETGRGID_R)
 
+check_c_source_compiles(
+    "
+    #include <asm/termbits.h>
+    #include <sys/ioctl.h>
+
+    int main(void)
+    {
+        struct termios2 t;
+        return 0;
+    }
+    "
+    HAVE_TERMIOS2)
```

Comment:
do we know which rids will not have this defined?

Reply:
Actually, we also need to update the eng/native/tryrun.cmake for cross build to get this definition. The live detection in `configure.cmake` is only for non-cross builds. So we need to get the result of this check on all our cross target platforms (by running the build is running natively there) and update the tryrun.cmake accordingly in this PR.
@krwq that should give an answer to your question then.

Reply:
According to https://man7.org/linux/man-pages/man2/ioctl_tty.2.html this was added in 2.6.20 Kernel e.g. it should be available in all Linux versions we support. (and is Linux specific) I saw note that it may not work on architectures but I don't know any details. The definition lives in `asm-generic/termbits.h` so I expect it would compile everywhere.

Reply:
Do we ever use it for OSes other than Linux? (Like FreeBSD ARM64)? I was looking for section with plain LINUX but aside from ALPINE I did not find good fit.

Reply:
The section that is used for linux starts here: https://github.com/dotnet/runtime/blob/4cf8c0743880e54ded0d7985d433bdbf52acbb12/eng/native/tryrun.cmake#L74

It is used for all cross builds, that means including FreeBSD. Since the section for Linux is shared with FreeBSD, Illumos and Tizen, we need to know the value for those OSes. It seems that it would be as easy as cross building the repo for those three OSes with your change and the tryrun.cmake modified. If it was not supported, the build would fail.

Reply:
I check FreeBSD headers and there is neither `termios2` or `TCGETS2`. I suspect others would be the same. 
Instead of setting negative value for all the other OSes, would it make sense to create new branch for Linux only? 
And if do we already have some variable that we can use (CMAKE_SYSTEM_NAME, CLR_CMAKE_TARGET_OS, ..?) or would we need to invent something?

Reply:
Since everything else is the same, I'd just make setting this single value conditional based on where it is supported.